v1.0.0 Midtrans-Go Revamp

Xaxxis opened the pull request:
Feature:
- Client for Iris, Snap, and CoreApi
- Can change default go http client
- Can set notification override, append
- Can set idempotency-key Payment and Iris
- Can request via map
- Can set globally logger level (error, info, debug)
- Can set context via client options

Commit:
Update .gitignore

Commit:
Create const for banks

Commit:
Create client implementation for Go HTTP Client

Commit:
Create core-api client/gateway implementation

Commit:
Create core-api test client/gateway

Commit:
Create core-api request and response struct

Commit:
Create Midtrans Error

Commit:
Implement go module

Commit:
Create Iris client/gateway implementation

Commit:
Create iris test client/gateway

Commit:
Create iris request and response struct

Commit:
Create log implementation

Commit:
Midtrans base configuration

Commit:
Payment source constant

Commit:
Midtrans base request struct

Commit:
Create snap client/gateway implementation

Commit:
Create snap test client/gateway

Commit:
Create snap request and response struct

Commit:
Go midtrans example

Commit:
Create Readme midtrans-go

Commit:
add function detail in each function/method

Commit:
QR-Change const prefix PaymentType from `Source` to `PaymentType`

Commit:
QR - Rename `Gateway` to `Client` and simplify New function on CoreApi

- add how to add options config in sample

Commit:
QR - Rename `Gateway` to `Client` and simplify New function on Iris

Commit:
QR - Rename `Gateway` to `Client` and simplify New function on Snap

- add sample how to add options config

Commit:
- Rename `getClient` function to `getDefaultClient`
- Fix function description for

Commit:
Create sample transaction action integration

Commit:
Add sample option idempotency on transaction action request

Commit:
Update logic handle notification

Commit:
QR - Update readme as quick review from slack

Commit:
Update readme Override/append notif

Commit:
QR2 - Separate struct response for Refund

Commit:
QR2 - Separate struct response for check transaction and get status b2b

- add integration test for check transaction API (Get status)
- add integration test for API Get status B2B

rizdaprasetya committed:
clarify readme and improve wording

Commit:
QR2 - Refactor struct response core-api

The response for charge request, approve, deny, capture,  expire, cancel API

Commit:
Merge pull request #1 from rizdaprasetya/patch-1

clarify readme and improve wording

Commit:
QR3 - Put metadata object on the request and response core-api

Commit:
QR3 - Add secure 3ds param, on Snap sample request

Commit:
Merge branch 'main' of https://github.com/Xaxxis/midtrans-go into main

Commit:
QR3 - Refactor notification sample

Commit:
QR3 - Refactor midtrans client struct and getClient function

- Rename Midtrans Client struct to HttpClient
- Rename and refactor function getClient to GetHttpClient

Commit:
Removed unusual code on integration test

Commit:
QR4 - Update snap param with 3ds and sample notification

Commit:
QR4 - Add metadata param to Snap request

Commit:
QR4 - Separated PaymentType for Snap and update sample

Commit:
QR4 - Separated PaymentType for CoreApi and update sample

Commit:
QR4 - Rename client.go to httpclient.go

Commit:
QR4 - Delete payment.go moved to PaymentType for Snap and coreapi

Commit:
Uncomment some test

rizdaprasetya reviewed (approved):
All review communicated internally has been adressed. Good to merge, please merge and release. Thanks! 👍

Commit:
QR5-Refactor coreapi Response struct to ChargeResponse and create type deny,approve,cancel,capture,expire

Commit:
Change variable name to string instead of empty interface

Commit:
Update core-api typo response struct name in Expire API

Commit:
QR6 = Change resp variable to nil for consistency code structure

Commit:
QR6 - Update customer details should omitempty as all attribute and move callback object to snap request

Commit:
QR6 - Update omitempty for some attribute

- recipient_name, BCA sub_company, GopayDetails attribute, acquirer Qris

Commit:
QR6 - Moved callback to snap request, Add omitempty for some attribute

- Cstore object attribute
- GopayDetails object attribute
- PermataVa object attribute
- BniVa object attribute
- BcaVa va_number attribute
- DynamicDescriptor object attribute
- InstallmentDetail object attribute

Commit:
Change resp variable in error block to nil for consistency code structure

Commit:
QR6 = Change nil variable in error block with variable for consistency code structure

Commit:
QR6 - Fix error handling in status code, add function HasOwnProperty

Commit:
Fix sample code and core-api client test

Commit:
Change typo on Readme
